scripts/num_components_paramsweep.py

```diff
@@ -0,0 +1,133 @@
+"""
+Gregory Way 2018
+scripts/num_components_param_sweep.py
+
+Usage: Run in command line:
+
+python scripts/num_components_param_sweep.py
+
+     with required command arguments:
+
+       --parameter_file     filepath pointing to tab separated parameters
+       --config_file        filepath pointing to PMACS configuration file
+
+     and optional arguments:
+
+       --algorithm          a string indicating which algorithm to sweep over
+                              default: 'tybalt' (i.e. variational autoencoder)
+       --python_path        absolute path of PMACS python in select environment
+                              default: '~/.conda/envs/tybalt-gpu/bin/python'
+       --param_folder       filepath of where to save the results
+                              default: 'param_sweep'
+
+Output:
+Submit a job to the PMACS cluster training a distinct VAE with a different
+combination of hyper parameters.
+"""
+
+import os
+import argparse
+import pandas as pd
+from bsub_helper import bsub_help
+
+
+def get_param(param):
+    try:
+        sweep = parameter_df.loc[param, 'sweep']
+    except:
+        sweep = ''
+        print("Warning! No parameter detected of name: {}".format(param))
+    return sweep.split(',')
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-p', '--parameter_file',
+                    help='location of tab separated parameter file to sweep')
+parser.add_argument('-c', '--config_file',
+                    help='location of the configuration file for PMACS')
+parser.add_argument('-a', '--algorithm',
+                    help='which algorithm to sweep hyperparameters over')
+parser.add_argument('-s', '--python_path',
+                    help='absolute path of python version',
+                    default='~/.conda/envs/tybalt-gpu/bin/python')
+parser.add_argument('-d', '--param_folder',
+                    help='folder to store param sweep results',
+                    default='param_sweep')
+parser.add_argument('-t', '--script',
+                    help='path the script to run the parameter sweep over',
+                    default='scripts/vae_pancancer.py')
+args = parser.parse_args()
+
+parameter_file = args.parameter_file
+config_file = args.config_file
+algorithm = args.algorithm
+python_path = args.python_path
+param_folder = args.param_folder
+script = args.script
+
+if not os.path.exists(param_folder):
+    os.makedirs(param_folder)
+
+# Load data
+parameter_df = pd.read_table(parameter_file, index_col=0)
+config_df = pd.read_table(config_file, index_col=0)
+
+# Retrieve hyperparameters to sweep over
+num_components = get_param('num_components')
```

Comment:
Are you relying on panda's string to number conversion for this block? Thats fine, but may want to be explicit on how one creates the parameter file.

Reply:
Sounds good. Right now I rely on scripts that ingest these parameters to perform the appropriate conversions. I bolstered the documentation in a recent commit.